ros2cli/ros2cli/command/__init__.py

```diff
@@ -201,7 +200,9 @@ def add_subparsers_on_demand(
         args = argv
         # for completion use the arguments provided by the argcomplete env var
         if _is_completion_requested():
-            args = shlex.split(os.environ['COMP_LINE'])[1:]
+            from argcomplete import split_line
```

Comment:
The big reason not to do this is that `argcomplete` is currently an optional dependency of `ros2cli`, and this makes it required.  Is there another way we can accomplish the same thing?

Reply:
I was wondering if there is a case that we cannot load argcomplete and _is_completion_requested True?

Reply:
I'm honestly not sure.  It looks like the definition of `_is_comletion_requested()` is:

```
def _is_completion_requested():
    return os.environ.get('_ARGCOMPLETE') == '1'
```

I don't know what is expected to set the `_ARGCOMPLETE` environment variable, so I'm not sure what situations get us here.

Reply:
`_ARGCOMPLETE` appears [here](https://github.com/kislyuk/argcomplete/blob/6d33c8e321ffa388c879fec56a80add0181f56b8/argcomplete/shell_integration.py#L44) in argcomplete registration script.

This is called from a ros2cli script, if the script is available meaning argcomplete available I believe,  
https://github.com/ros2/ros2cli/blob/df55a4ed42cdaca1fa103bb1b9807ce2c239055f/ros2cli/completion/ros2-argcomplete.bash#L15-L19

which is invoked when you call setup script
https://github.com/ros2/ros2cli/blob/e5f67d0e0362462644a5530094fac43af84ca938/ros2cli/colcon.pkg#L1-L6

Reply:
Thanks for pointing it out.  This does look OK then.